functions/Remove-DbaDbSequence.ps1

```diff
@@ -57,42 +60,74 @@ function Remove-DbaDbSequence {
-            return
-        }
+    begin {
+        $sequences = @( )
+    }
 
-        foreach ($instance in $SqlInstance) {
-            $InputObject += Get-DbaDatabase -SqlInstance $instance -SqlCredential $SqlCredential -Database $Database
+    process {
+        if ($SqlInstance) {
+            $params = $PSBoundParameters
+            $null = $params.Remove('WhatIf')
+            $null = $params.Remove('Confirm')
+            $sequences = Get-DbaDbSequence @params
+        } else {
+            $sequences += $InputObject
         }
+    }
 
-        foreach ($db in $InputObject) {
-
-            if ($Pscmdlet.ShouldProcess($db.Parent.Name, "Removing the sequence $Name in the $Schema schema in the database $($db.Name) on $($db.Parent.Name)")) {
+    end {
+        # We have to delete in the end block to prevent "Collection was modified; enumeration operation may not execute." if directly piped from Get-DbaDbSequence.
+        foreach ($sequence in $sequences) {
+            if ($PSCmdlet.ShouldProcess($sequence.Parent.Parent.Name, "Removing the sequence $($sequence.Schema).$($sequence.Name) in the database $($sequence.Parent.Name) on $($sequence.Parent.Parent.Name)")) {
+                $output = [pscustomobject]@{
+                    ComputerName   = $sequence.Parent.Parent.ComputerName
+                    InstanceName   = $sequence.Parent.Parent.ServiceName
+                    SqlInstance    = $sequence.Parent.Parent.DomainInstanceName
+                    Database       = $sequence.Parent.Name
+                    Sequence       = "$($sequence.Schema).$($sequence.Name)"
+                    SequenceName   = $sequence.Name
+                    SequenceSchema = $sequence.Schema
+                    Status         = $null
+                    Removed        = $false
+                    Success        = $false
+                    Successful     = $false
+                }
                 try {
-                    $sequenceToDrop = $db.Sequences | Where-Object { $_.Schema -eq $Schema -and $_.Name -eq $Name }
-                    $sequenceToDrop.Drop()
+                    $sequence.Drop()
+                    $output.Status = "Dropped"
+                    $output.Removed = $true
+                    $output.Success = $true
+                    $output.Successful = $true
                 } catch {
-                    Stop-Function -Message "Failure on $($db.Parent.Name) to drop the sequence $Name in the $Schema schema in the database $($db.Name)" -ErrorRecord $_ -Continue
+                    Stop-Function -Message "Failed removing the sequence $($sequence.Schema).$($sequence.Name) in the database $($sequence.Parent.Name) on $($sequence.Parent.Parent.Name)" -ErrorRecord $_
+                    $output.Status = (Get-ErrorMessage -Record $_)
+                    $output.Removed = $false
+                    $output.Success = $false
+                    $output.Successful = $false
```

Comment:
Why the two properties? `Success` and `Successful`

Reply:
To show you, the other contributors and @potatoqualitee , that we have some commands that output "Success" and some output "Successful". So I (temporary) put both there and also the "Removed" that is used in commands as well to start a discussion about how to name the (one) parameter, that should indicate the successful removal of the object and that can be testet in the surrounding code. I think this parameter should be named the same in all Remove-Dba* commands.

Reply:
Well the use of a property to indicate success in my opinion is not required at all.

But the normal pattern is not used in the module 😁. I'd vote for `IsRemoved` so it is unique to the object. The success properties are for things like Copy commands that are more complex and need the same property across multiple commands.

Reply:
Yes, I like IsRemoved as well.

Reply:
@potatoqualitee you may have the last word on how to name the output properties, I will then do the changes.